src/main/java/org/ohdsi/circe/cohortdefinition/builders/BuilderUtils.java

```diff
@@ -25,14 +26,17 @@ public static String getCodesetJoinExpression(Integer standardCodesetId, String
         String joinExpression = "";
         ArrayList<String> codesetClauses = new ArrayList<>();
         String excludeExpression = exclude ? "<>" : "=";
+        String aliasPostfix = StringUtils.replaceAll(UUID.randomUUID().toString(), "-", "");
```

Comment:
I think you're creating this in order to support multiple joins to the `#codeset` table, correct?  I was thinking you could do an '`in` clause instead of adding joins.  As in:

```
WHERE  specialty_concept_id in (select concept_id from #codeset where codeset_id = @specialtyConceptId)
```

I think DBMSs can perform the necessary optimization where the `in()` is optimized to the same work as a join.  The reason I like this form is that the criteria are all applied together in the WHERE clause and you don't have it mixed between JOINs and WHEREs.   

I'm assuming that you followed the `JOIN` pattern from how we use the codeset_id when querying the domain tables (like for condition occurrence) but that was because, at the time, it wasn't an optional thing so putting it as part of the 'core' query made sense.  But in this case, I think it would 'look' better if we just appended to the WHERE clauses (where concept_id in (select...) unless we can demonstrate a performance impact.